### ISSUE TITLE
Cannot build CPU version

### ISSUE THREAD

**Issue by marcus-nilis** [openxla/xla#4207]
Could someone please help with initial build? Manual seems to be outdated https://github.com/openxla/xla/blob/main/docs/developer_guide.md

Repro steps:

```
git clone https://github.com/openxla/xla.git
cd xla
docker pull tensorflow/build:latest-python3.9
docker run --name xla -w /xla -it -d --rm -v $PWD:/xla tensorflow/build:latest-python3.9 bash
docker exec xla ./configure
docker exec xla bazel build --test_output=all --spawn_strategy=sandboxed //xla/...
```

1 hour later: 

```
ERROR: /xla/xla/tools/BUILD:160:14: Linking xla/tools/dumped_computation_to_text failed: (Exit 1): gcc failed: error executing command (from target //xla/tools:dumped_computation_to_text) /usr/bin/gcc @bazel-out/k8-opt/bin/xla/tools/dumped_computation_to_text-2.params

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
bazel-out/k8-opt/bin/xla/backends/interpreter/_objs/platform/platform.o:platform.cc:function stream_executor::interpreter::XlaInterpreterPlatform::GetExecutor(stream_executor::StreamExecutorConfig const&): error: undefined reference to 'stream_executor::ExecutorCache::GetOrCreate(stream_executor::StreamExecutorConfig const&, std::function<absl::lts_20230125::StatusOr<std::unique_ptr<stream_executor::StreamExecutor, std::default_delete<stream_executor::StreamExecutor> > > ()> const&)'
bazel-out/k8-opt/bin/xla/backends/interpreter/_objs/platform/platform.o:platform.cc:function std::_Rb_tree<int, std::pair<int const, stream_executor::ExecutorCache::Entry>, std::_Select1st<std::pair<int const, stream_executor::ExecutorCache::Entry> >, std::less<int>, std::allocator<std::pair<int const, stream_executor::ExecutorCache::Entry> > >::_M_erase(std::_Rb_tree_node<std::pair<int const, stream_executor::ExecutorCache::Entry> >*): error: undefined reference to 'stream_executor::ExecutorCache::Entry::~Entry()'
collect2: error: ld returned 1 exit status

```
Tried to run it manually from the container along with 'bazel clean --expunge' - no luck.

**Comment by SunDoge** [comment]
I met a similar problem when building GPU version,

```
ERROR: /xla/xla/tools/BUILD:160:14: Linking xla/tools/dumped_computation_to_text failed: (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command (from target //xla/tools:dumped_computation_to_text) external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc @bazel-out/k8-opt/bin/xla/tools/dumped_computation_to_text-2.params

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
/usr/bin/ld: bazel-out/k8-opt/bin/xla/backends/interpreter/libplatform.lo(platform.o): in function `stream_executor::interpreter::XlaInterpreterPlatform::GetExecutor(stream_executor::StreamExecutorConfig const&)':
platform.cc:(.text._ZN15stream_executor11interpreter22XlaInterpreterPlatform11GetExecutorERKNS_20StreamExecutorConfigE+0x4c): undefined reference to `stream_executor::ExecutorCache::GetOrCreate(stream_executor::StreamExecutorConfig const&, std::function<absl::lts_20230125::StatusOr<std::unique_ptr<stream_executor::StreamExecutor, std::default_delete<stream_executor::StreamExecutor> > > ()> const&)'
/usr/bin/ld: bazel-out/k8-opt/bin/xla/backends/interpreter/libplatform.lo(platform.o): in function `std::_Rb_tree<int, std::pair<int const, stream_executor::ExecutorCache::Entry>, std::_Select1st<std::pair<int const, stream_executor::ExecutorCache::Entry> >, std::less<int>, std::allocator<std::pair<int const, stream_executor::ExecutorCache::Entry> > >::_M_erase(std::_Rb_tree_node<std::pair<int const, stream_executor::ExecutorCache::Entry> >*)':
platform.cc:(.text._ZNSt8_Rb_treeIiSt4pairIKiN15stream_executor13ExecutorCache5EntryEESt10_Select1stIS5_ESt4lessIiESaIS5_EE8_M_eraseEPSt13_Rb_tree_nodeIS5_E[_ZNSt8_Rb_treeIiSt4pairIKiN15stream_executor13ExecutorCache5EntryEESt10_Select1stIS5_ESt4lessIiESaIS5_EE8_M_eraseEPSt13_Rb_tree_nodeIS5_E]+0x35): undefined reference to `stream_executor::ExecutorCache::Entry::~Entry()'
collect2: error: ld returned 1 exit status
INFO: Elapsed time: 3323.122s, Critical Path: 277.33s
INFO: 17041 processes: 5146 internal, 11895 processwrapper-sandbox.
FAILED: Build did NOT complete successfully
```

**Comment by Alwaysproblem** [comment]
Hello, I encountered a similar issue. I discovered that the `if_static` function within the `xla/stream_executor/BUILD:753` has an impact on the compilation failure of the `dumped_computation_to_text` target.

To resolve this, I attempted removing the `if_static` function from the `BUILD` file, and this approach proved effective. However, I am uncertain whether this modification might have any unintended consequences on the functionality of XLA.

So, I managed to successfully compile using a static library with the following command:

```shell
bazel build --define=framework_shared_object=1 --test_output=all --spawn_strategy=sandboxed //xla/...
```

This command yielded a successful build.
